### PR TITLE
Port templating to RHEL 8

### PR DESCRIPTION
**Pull request by VladimirSlavik**
Same as #4402 but for -8.

### CODE REVIEW

**Review by rvykydal — APPROVED** [review]
Thank you.

**Comment by VladimirSlavik** [comment]
@jstodola can you please check that this is ok? There's a lot of changes outside the infra definition...

File: .github/workflows/infra-check.yml [hunk 1]
```diff
@@ -71,3 +72,54 @@ jobs:
...
+            git cherry-pick "$COMMIT"
+            make -f Makefile.am reload-infra
+            CHANGES=$(git status -s)
+            if [[ -n $CHANGES ]] ; then
+              echo "Templates out of sync after commit $COMMIT:"
+              git log -1 "$COMMIT"
+              git status
+              exit 1
+            fi
+          done
```
Comment (jkonecny12) [inline .github/workflows/infra-check.yml:125]: Same as in RHEL-9 PR. Merging these could simplify this.

Also I'm thinking about adding `set -eu` to make it more "robust".

**Review by jkonecny12 — APPROVED** [review]
Looks good to me.

However, I'm thinking that we should also add:
https://github.com/rhinstaller/anaconda/blob/rhel-8/dockerfile/anaconda-ci/rhel-8.repo
and all the `dockerfile/*/rhel*.repo` because these should point to the correct minor to avoid testing 8.8 on rhel-8.7 branch etc.
It should stay latest on the `rhel-8` or `rhel-9` branches but should change on minor. Probably a new variable or something calculated from the distribution one?

Could be resolved as followup if you want.

**Review by jstodola — APPROVED** [review]

**Pull request merged**